confluent_kafka/schema_registry/avro.py

```diff
@@ -0,0 +1,227 @@
+import json
+import struct
+from copy import deepcopy
+
+from fastavro import parse_schema, schemaless_writer, schemaless_reader
+
+from confluent_kafka.schema_registry import MessageField, TopicNameStrategy, _MAGIC_BYTE, \
+    SubjectNameStrategy, ContextStringIO
+from confluent_kafka.serialization import Serializer, Deserializer, SerializationError
+
+
+class AvroSerializer(Serializer):
+    """
+    AvroSerializer encodes objects into the Avro binary format in accordance
+    with the schema provided at instantiation.
+
+    Unless configured otherwise the AvroSerializer will automatically register
+    its schema with the provided registry instance.
+
+    Args:
+        conf (dict): Avro Serializer configuration instance.
+        schema_registry_client (SchemaRegistryClient): Schema Registry client.
+        schema (Schema): Schema instance to use when encoding Avro records.
+
+    .. _Schema Registry specification
+        https://avro.apache.org/docs/current/spec.html for more details.
+
+    .. _Schema Resolution
+        https://avro.apache.org/docs/1.8.2/spec.html#Schema+Resolution
+
+    """
+
+    def __init__(self, conf, schema_registry_client, schema):
+        self.schema = schema
+        self.schema_id = None
+        self.parsed_schema = parse_schema(self._prepare_schema(schema))
+        self.key_subject_func, self.value_subject_func = self.handle_config(conf)
+
+        self.registry = schema_registry_client
+        self.auto_register = True
+        self.known_subjects = {}
+
+    def handle_config(self, src_conf):
+        """
+        Validates AvroSerializer properties.
+
+        Args:
+            src_conf(dict): source configuration
+
+        Returns:
+            SubjectNameStrategy: Key subject name strategy implementation.
+            SubjectNameStrategy: Value subject name strategy implementation.
+
+        """
+        conf = deepcopy(src_conf)
+
+        self.auto_register = src_conf.get('auto.register.schema', True)
+
+        if not isinstance(self.auto_register, bool):
+            raise ValueError("auto.register.schema must be a boolean value")
+
+        key_subject_name_strategy = conf.pop('key.subject.name.strategy',
+                                             TopicNameStrategy())
```

Comment:
don't necessarily have a problem with this, but i think it needs to be thought through how having object instances that don't directly translate to string values interpoerates with standard python config file interop stuff, whatever that is.

Reply:
also even non string values (like `True`)